src/fides/api/models/privacy_experience.py

```diff
@@ -506,17 +550,11 @@ def upsert_privacy_experiences_after_config_update(
     ] = remove_config_from_matched_experiences(db, experience_config, removed_regions)
 
     for region in regions:
-        (
-            overlay_experience,
-            privacy_center_experience,
-        ) = PrivacyExperience.get_experiences_by_region(db, region.value)
-
-        existing_experience: Optional[PrivacyExperience] = (
-            overlay_experience
-            if experience_config.component == ComponentType.overlay
-            else privacy_center_experience
+        existing_experience: Optional[
+            PrivacyExperience
+        ] = PrivacyExperience.get_experience_by_region_and_component(
+            db=db, region=region, component=experience_config.component  # type: ignore[arg-type]
         )
```

Comment:
This method updates existing Experiences when ExperienceConfigs (the copy) have new regions added to them.  It assumed that only an overlay and a privacy center experience existed, but we've added a new type.